src/Utils.ts

```diff
@@ -167,7 +168,19 @@ const Utils = {
             cqPath = normalizePath(cqPath);
         }
         return cqPath;
-    }
+    },
+
+    fetchModelOnBackNav(componentProps: ComponentProps): void {
+        const { cqPath, history } = componentProps;
+
+        window.onpopstate = () => {
```

Comment:
Shouldn't the fix be here ?
https://github.com/adobe/aem-spa-page-model-manager/blob/master/src/ModelRouter.ts#L190

Reply:
Ah yes, that is cleaner. Thanks, let me try to rearrange the code.